controllers/pipelinerun_controller.go

```diff
@@ -0,0 +1,205 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	hasv1alpha1 "github.com/redhat-appstudio/application-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/api/v1alpha1"
+	"github.com/redhat-appstudio/release-service/helpers"
+	"github.com/redhat-appstudio/release-service/tekton"
+	tektonv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"knative.dev/pkg/apis"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// PipelineRunReconciler reconciles a PipelineRun object
+type PipelineRunReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=pipelineruns,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=pipelineruns/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=appstudio.redhat.com,resources=pipelineruns/finalizers,verbs=update
+
+// Reconcile is part of the main kubernetes reconciliation loop which aims to
+// move the current state of the cluster closer to the desired state.
+func (r *PipelineRunReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("PipelineRun", req.NamespacedName)
+
+	pipelineRun := &tektonv1beta1.PipelineRun{}
+	err := r.Get(ctx, req.NamespacedName, pipelineRun)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Info("PipelineRun resource not found")
+
+			return ctrl.Result{}, nil
+		}
+
+		log.Error(err, "Failed to get PipelineRun")
+
+		return ctrl.Result{}, err
+	}
+
+	if tekton.IsBuildPipelineRun(pipelineRun) {
+		return r.createRelease(ctx, pipelineRun)
+	} else if tekton.IsReleasePipelineRun(pipelineRun) {
+		return r.updateReleaseStatus(ctx, pipelineRun)
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// createRelease creates a new Release by using the information provided in the Build PipelineRun.
+func (r *PipelineRunReconciler) createRelease(ctx context.Context, pipelineRun *tektonv1beta1.PipelineRun) (ctrl.Result, error) {
+	log := r.Log.WithValues()
+
+	log.Info("Creating new Release",
```

Comment:
What's the advantage to logging like this instead of passing it as arguments to Log.WithValues like we did before? Or, what is the difference, rather

Reply:
In this case I wanted to print different things. In some logs I wanted the PipelineRun but in others the component.